### PR TITLE
add CHD create

### PR DESCRIPTION
**Pull request by Lothrien**
create chds

### CODE REVIEW

File: SabreTools.DatTools/Rebuilder.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 using System.Collections.Generic;
+using System;
```
Comment (mnadareski) [inline SabreTools.DatTools/Rebuilder.cs:2]: Ordering, again. For both new `using` statements.

File: SabreTools.FileTypes/BaseFile.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@
 using SabreTools.Core;
 using SabreTools.Core.Tools;
 using SabreTools.FileTypes.Aaru;
-using SabreTools.FileTypes.CHD;
+//using SabreTools.FileTypes.CHD;
```
Comment (mnadareski) [inline SabreTools.FileTypes/BaseFile.cs:9]: Remove, don't comment out.

File: SabreTools.FileTypes/BaseFile.cs [hunk 3]
```diff
@@ -62,6 +62,11 @@ public class BaseFile
         /// </summary>
         public string Parent { get; set; }
 
+        /// <summary>
+        /// Childs of the file
```
Comment (mnadareski) [inline SabreTools.FileTypes/BaseFile.cs:66]: "Children" not "Childs"

File: SabreTools.FileTypes/BaseFile.cs [hunk 4]
```diff
@@ -62,6 +62,11 @@ public class BaseFile
...
+        /// </summary>
+        public string Child { get; set; }
```
Comment (mnadareski) [inline SabreTools.FileTypes/BaseFile.cs:68]: "Children" not "Child". You probably also want a list or array of children names, not just a single `string`.

File: SabreTools.FileTypes/BaseFile.cs [hunk 5]
```diff
@@ -304,8 +309,8 @@ public static BaseFile GetInfo(string input, string header = null, Hash hashes =
             BaseFile baseFile;
             if (fileType == FileType.AaruFormat && !asFiles.HasFlag(TreatAsFile.AaruFormat))
                 baseFile = AaruFormat.Create(inputStream);
-            else if (fileType == FileType.CHD && !asFiles.HasFlag(TreatAsFile.CHD))
-                baseFile = CHDFile.Create(inputStream);
+//            else if (fileType == FileType.CHD && !asFiles.HasFlag(TreatAsFile.CHD))
```
Comment (mnadareski) [inline SabreTools.FileTypes/BaseFile.cs:312]: I don't like this being removed

File: SabreTools.FileTypes/CHD/CHDFile.cs [hunk 6]
```diff
@@ -1,16 +1,19 @@
-﻿using System;
+﻿using System.Collections.Generic;
+using System;
```
Comment (mnadareski) [inline SabreTools.FileTypes/CHD/CHDFile.cs:2]: Ordering of using statements.

File: SabreTools.FileTypes/CHD/CHDFile.cs [hunk 7]
```diff
@@ -1,16 +1,19 @@
...
 
+// TODO: add check
 namespace SabreTools.FileTypes.CHD
 {
     /// <summary>
     /// This is code adapted from chd.h and chd.cpp in MAME
     /// Additional archival code from https://github.com/rtissera/libchdr/blob/master/src/chd.h
     /// </summary>
-    public abstract class CHDFile : BaseFile
+    public class CHDFile : BaseArchive
```
Comment (mnadareski) [inline SabreTools.FileTypes/CHD/CHDFile.cs:16]: Why is this no longer `abstract`? Each of the versioned CHDs still need to be individually accounted for.

File: SabreTools.FileTypes/CHD/CHDFile.cs [hunk 8]
```diff
@@ -19,33 +22,62 @@ public abstract class CHDFile : BaseFile
         protected uint length;              // length of header (including tag and length fields)
         protected uint version;             // drive format version
 
+        protected CHDFile generated;
```
Comment (mnadareski) [inline SabreTools.FileTypes/CHD/CHDFile.cs:25]: You don't have any comments on here so I don't understand what either of these do.

**Review by mnadareski — COMMENTED** [review]
As with the ISO/HDD one, I'm hesitant to add this as functionality to SabreTools. It doesn't really match what is normally expected for use.

File: SabreTools.FileTypes/CHD/CHDFile.cs (hunk 7 above)
Comment (Lothrien) [reply SabreTools.FileTypes/CHD/CHDFile.cs:16]: its now a basearchive, more treated like an archive not a file. Oportunity is to transform in both directions.

File: SabreTools.FileTypes/CHD/CHDFile.cs (hunk 8 above)
Comment (Lothrien) [reply SabreTools.FileTypes/CHD/CHDFile.cs:25]: yeah, that code be done better - i had some problems with giving arguments and keeping them inside...

File: SabreTools.FileTypes/BaseFile.cs (hunk 4 above)
Comment (Lothrien) [reply SabreTools.FileTypes/BaseFile.cs:68]: ohh, yep :(

File: SabreTools.FileTypes/BaseFile.cs (hunk 2 above)
Comment (Lothrien) [reply SabreTools.FileTypes/BaseFile.cs:9]: WIP

**Comment by mnadareski** [comment]
How about this, for CHD, HDD, and ISO support, can you either:
1. Make this a setting (or set of settings) that allow for turning on this functionality?
or
2. Make this an entirely separate feature to convert between the 3 formats?

If you're set on including this work, I'm leaning more heavily toward the second option, since it would be less likely to affect normal users in ways that they would not expect.

I also would like to request that you either make a new branch for the combined work or at least combine the work you've done into a single existing branch. Trying to review these PRs in tandem and trying to figure out how multiple would work together is rather difficult.

**Pull request closed without merging**